src/accountsdb/db.zig

```diff
@@ -434,61 +416,54 @@ pub const AccountsDB = struct {
         defer reference_memory_lg.unlock();
-        try reference_memory.ensureTotalCapacity(@intCast(file_names.len));
+        try reference_memory.ensureTotalCapacity(@intCast(fields.file_map.count()));
 
-        for (file_names, 1..) |file_name, file_count| {
-            // parse "{slot}.{id}" from the file_name
-            var fiter = std.mem.tokenizeSequence(u8, file_name, ".");
-            const slot = std.fmt.parseInt(Slot, fiter.next().?, 10) catch |err| {
-                self.logger.warnf("failed to parse slot from {s}", .{file_name});
-                return err;
-            };
-            const file_id_usize = try std.fmt.parseInt(usize, fiter.next().?, 10);
-
-            // read metadata
-            const file_infos: ArrayList(AccountFileInfo) = self.fields.?.file_map.get(slot) orelse {
-                // dont read account files which are not in the file_map
-                // note: this can happen when we load from a snapshot and there are extra account files
-                // in the directory which dont correspond to the snapshot were loading
-                self.logger.warnf("failed to read metadata for slot {d}", .{slot});
-                continue;
-            };
+        for (
+            fields.file_map.keys()[file_map_start_index..file_map_end_index],
+            fields.file_map.values()[file_map_start_index..file_map_end_index],
+            1..,
+        ) |slot, file_infos, file_count| {
             // if this is hit, its likely an old snapshot
-            if (file_infos.items.len != 1) {
+            if (file_infos.len != 1) {
                 self.logger.errf("incorrect file_info count for slot {d}, likley trying to load from an unsupported snapshot\n", .{slot});
                 return error.InvalidSnapshot;
             }
-            const file_info = file_infos.items[0];
-            if (file_info.id != file_id_usize) {
-                self.logger.errf("file_id from metadata for slot {d} doesnt match file_id from filename: {d} vs {d}\n", .{ slot, file_info.id, file_id_usize });
-                return error.InvalidSnapshotFormat;
-            }
+            const file_info = file_infos[0];
 
             // read accounts file
-            const abs_path = try std.fmt.bufPrint(&buf, "{s}/{s}", .{ accounts_dir_path, file_name });
-            const accounts_file_file = try std.fs.cwd().openFile(abs_path, .{ .mode = .read_write });
-            var accounts_file = AccountFile.init(accounts_file_file, file_info, slot) catch |err| {
-                self.logger.errf("failed to *open* AccountsFile {s}: {s}\n", .{ file_name, @errorName(err) });
-                return err;
+            var accounts_file = blk: {
+                const file_name_bounded = sig.utils.fmt.boundedFmt(
+                    "{d}.{d}",
+                    .{ slot, file_info.id },
+                    .{ std.math.maxInt(Slot), std.math.maxInt(@TypeOf(file_info.id)) },
+                ) catch unreachable;
```

Comment:
Technically there's a way to validate this kind of thing using type info at comptime, eliding the runtime error, thing is it's a fair bit more complicated and a bit less general.

In this case we can obviously see the error is unreachable, since the arguments could never overflow based on the max args. I can look into doing this anyway if you want, lmk if you think the tradeoff is still worth it.

Reply:
rn we use this function for only integers - what if we only support integers which would make it comptime checkable and we could remove the max_args parameter - then if we want to add more type support we can add as we need - wdyt?

Reply:
Sounds like a fair compromise - the alternative use case I was thinking of was a situation where we had a variable-length string with a comptime-known upper bound, there doesn't exist a neat way to express that kind of thing only through types.

Although, maybe such a use case could be supported by requiring non-integer types to expose a decl like `pub const max_len: usize = value;`, so then you could wrap the string with the upper bound in a type which exposes that upper bound.

I'll go ahead with the former, lmk if you're for or against the latter.

Reply:
yeah probably better to go with the simpler approach for now and add a comment which describes the latter soln for future work if we need it

Reply:
I think the code as is is relatively self-descriptive. Best case scenario is we never actually need to use this for anything other than printing integers - if a more complex use case arises, we can adapt the code as needed without limiting the design space right now.